### PR TITLE
[packaging] Deprecate old dd-agent, dd-forwarder & dogstatsd commands

### PR DESCRIPTION
**Pull request by degemer**
And some pep8 cleaning.

### CODE REVIEW

File: agent.py [hunk 1]
```diff
@@ -236,6 +237,12 @@ def main():
         sys.stderr.write("Unknown command: %s\n" % command)
         return 3
 
+    # Deprecation notice
+    if command not in DD_AGENT_COMMANDS:
+        # Will become an error message and exit after deprecation period
+        from utils.deprecations import deprecate_old_command_line_tools
+        deprecate_old_command_line_tools(__file__)
+
```
Comment (LeoCavaille) [inline agent.py]: Maybe this could be more granular as we're only deprecating certain subcommands and not `jmx, `check` and `flare` ?

Comment (LeoCavaille) [inline agent.py]: Ok misread this code. Fuggedaboutit

**Comment by LeoCavaille** [comment]
Nice, sorry for misreading this at first.
Looks good to me, thanks for organizing the code and the commits properly! It was an easy read :100:

**Comment by degemer** [comment]
Will rebase when 5.3 is out.

File: packaging/centos/datadog-agent.init [hunk 2]
```diff
@@ -27,7 +27,7 @@ AGENTCONF="/etc/dd-agent/datadog.conf"
 DOGSTATSDPATH="/opt/datadog-agent/agent/dogstatsd.py"
 AGENTUSER="dd-agent"
 PIDPATH="/var/run/dd-agent/"
-FORWARDERPATH="/usr/bin/dd-forwarder"
+FORWARDERPATH="/opt/datadog-agent/agent/ddagent.py"
```
Comment (LeoCavaille) [inline packaging/centos/datadog-agent.init]: Are we sure these scripts are made executable in our package?

File: utils/deprecations.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import logging
+from os.path import basename
+
+log = logging.getLogger(__name__)
```
Comment (LeoCavaille) [inline utils/deprecations.py]: Where is this logger outputting messages? I don't know is there is a `utils.` handler setup or a catchall, did you test it?

File: utils/deprecations.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def deprecate_old_command_line_tools(file_name):
+    name = basename(file_name)
+    if name == 'dd-forwarder' or name == 'dogstatsd' or name == 'dd-agent':
```
Comment (LeoCavaille) [inline utils/deprecations.py]: These match the command names (`basename(sys.argv[0])`) but not the file names (`basename(__file__)`) ?

File: packaging/centos/datadog-agent.init (hunk 2 above)
Comment (degemer) [inline packaging/centos/datadog-agent.init]: They are already `755` in the repo, and are only [copied](https://github.com/DataDog/dd-agent-omnibus/blob/master/config/software/datadog-agent.rb#L30) during the build, and [`chown`](https://github.com/DataDog/dd-agent-omnibus/blob/master/package-scripts/datadog-agent/postinst#L54) during the postinstall. They should stay `a+x` unless the build machine has a weird `umask`.

File: utils/deprecations.py (hunk 3 above)
Comment (degemer) [inline utils/deprecations.py]: I don't really know how the `logger` works, but it works, and seems to catch the right handler:
`dd-agent status`
> 2015-04-27 18:16:28,589 | WARNING | dd.collector | utils.deprecations(deprecations.py:10) | Using this command is deprecated and will be removed in a future version, for more information see https://github.com/DataDog/dd-agent/wiki/Deprecation-notice--(old-command-line-tools)

`dogstatsd stop`
> 2015-04-27 18:20:43,344 | WARNING | dd.dogstatsd | utils.deprecations(deprecations.py:10) | Using this command is deprecated and will be removed in a future version, for more information see https://github.com/DataDog/dd-agent/wiki/Deprecation-notice--(old-command-line-tools)

File: utils/deprecations.py (hunk 4 above)
Comment (degemer) [inline utils/deprecations.py]: Yep, it worked, but I changed it since it seems that `__file__` is not reliable. Thanks!

**Comment by LeoCavaille** [comment]
:+1:

**Pull request merged**